[yanqiyqh] iP

yanqiyqh opened the pull request:
# A Heading

## A Bullet List
- Item 1
- **Item 2**
- _Item 3_
  - Subitem 1
  - Subitem 2

## A Numbered List
1. First item
2. Second item
3. Third item

## A Fenced Code Block (with syntax highlighting)
```java
public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello, World!");
    }
}
```

## Task List
- [x] Complete the project documentation
- [x] Review the code for the latest features
- [ ] Deploy the application to production

## An Emoji
Here’s a thumbs up emoji: 👍

## A Blockquote
> "The only way to do great work is to love what you do."  
> — Steve Jobs

## A Hyperlink
Check out [GitHub](https://github.com) for more information.

## Inline Code
You can use the `print()` function in Python to display output.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yanqiyqh committed:
Level-0 changes

yanqiyqh committed:
Echo

yanqiyqh committed:
add and list

yanqiyqh committed:
added mark as done feature

yanqiyqh committed:
Added different types of tasks

yanqiyqh committed:
some minor changes

yanqiyqh committed:
basic error handling

yanqiyqh committed:
Added delete functionality

yanqiyqh committed:
added enum for support

yanqiyqh committed:
modify Main for save feature

yanqiyqh committed:
change to save commands instead of task history

yanqiyqh committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091' into branch-level-7

yanqiyqh committed:
minor changes

yanqiyqh committed:
add TaskList

yanqiyqh committed:
modify Ynch chatbot class to Ui-only

yanqiyqh committed:
modify YnchUi to have void methods for printing

yanqiyqh committed:
modify TaskList and delete Ynch

yanqiyqh committed:
add toSaveAsString method for file saving

yanqiyqh committed:
add toSaveAsString method

yanqiyqh committed:
add constructor for status

yanqiyqh committed:
add constructor for TaskList

yanqiyqh committed:
add size() method

yanqiyqh committed:
correct errors

yanqiyqh committed:
modify for More OOP

yanqiyqh committed:
modify toSaveAsString() method

yanqiyqh committed:
fix saving issue

yanqiyqh committed:
modify to LocalDate

yanqiyqh committed:
print in different format

yanqiyqh committed:
package into ynch.ui

yanqiyqh committed:
add getDescription() method

yanqiyqh committed:
add JUNIT test

yanqiyqh committed:
upgrade gradle version to 8.8

yanqiyqh committed:
add javadoc for Task-related classes

yanqiyqh committed:
add javadoc for Storage, TaskList, Parser and YnchUi

yanqiyqh committed:
modify size() method name

yanqiyqh committed:
tweak to follow coding standard

yanqiyqh committed:
add printFind and find methods

yanqiyqh committed:
fix bugs

yanqiyqh committed:
add find function

yanqiyqh committed:
resolve merge conflict

yanqiyqh committed:
resolve merge conflict

Potatoad88 commented on src/main/java/ynch/ui/Main.java:
```diff
@@ -0,0 +1,125 @@
+                break;
+            }
+            case ValidCommand.mark: {
+                int i = Integer.valueOf(userInput.split(" ")[1]);
```
Should the variable be named 'index' instead of 'i'?

Potatoad88 commented on src/main/java/ynch/ui/Main.java:
```diff
@@ -0,0 +1,125 @@
+        }
+    }
+
+    private static boolean isValidInput(String userInput) { 
```
Should the functions have javadoc comments before them?

Potatoad88 commented on src/main/java/ynch/ui/Task.java:
```diff
@@ -0,0 +1,109 @@
+
+class Task {
+    String task;
+    boolean isDone;
```
I like that you prefixed a boolean variable with 'is'☺️

Potatoad88 commented on src/main/java/ynch/ui/Main.java:
```diff
@@ -0,0 +1,125 @@
+        Storage storage = new Storage(filePath, filename);
+        TaskList taskList = storage.load();
+        Parser parser = new Parser();
+        YnchUi ui = new YnchUi();
```
I like that all your variable names are straightforward, it tells me exactly what they are for👍

Potatoad88 reviewed (commented):
Overall LGTM! 😃

yanqiyqh committed:
add GUI in progress

yanqiyqh committed:
add GUI

yanqiyqh committed:
add basic working GUI

yanqiyqh committed:
Add image resources DaUser.jpg and DaYnch.jpg

Added two new image resources:
  1. DaUser.jpg: This image will be used for the user's profile picture section.
  2. DaYnch.jpg: This image will serve as the default avatar for unregistered users.

- The images are located in the 'assets/images' directory and have been optimized for web performance.
- Future updates may include alt text for accessibility or different resolutions for responsive designs.

This commit is part of the ongoing effort to enhance the visual presentation of user profiles.

yanqiyqh committed:
feat: Add getTasks method to TaskList class

- Implemented the getTasks method to retrieve the current list of tasks.
- The method returns an ArrayList of Task objects representing the tasks in the to-do list.
- Ensures encapsulation by providing controlled access to the internal task list without directly exposing the underlying data structure.
- If no tasks are present in the list, an empty ArrayList is returned, preventing null pointer exceptions.

This commit enhances the TaskList class by providing a method for external access to the task list, improving usability and maintainability.

yanqiyqh committed:
Modify to keep up to date with Storage class' changes

yanqiyqh committed:
Add assertions to validate image loading in onStartUp method

- Added assertions to the `onStartUp` method in `MainWindow.java` to ensure that the images `DaUser.jpg` and `DaYnch.jpg` are correctly loaded.
- Assertions check that `userImage` and `ynchImage` are not null, helping to identify issues with image loading during startup.
- The assertions provide error messages if the images fail to load, improving debugging and reliability of the application startup process.
- Updated the method to handle potential image loading issues more effectively by validating image resources before adding them to the dialog container.

This commit enhances the robustness of the `onStartUp` method by ensuring critical resources are loaded correctly and providing immediate feedback in case of failures.

yanqiyqh committed:
Refactor Storage.java to have a cleaner switch case in stringToTask method

yanqiyqh committed:
Refactor Parser.java to use cleaner switch case

yanqiyqh committed:
Merge pull request #2 from yanqiyqh/branch-A-Assertions

Branch a assertions

yanqiyqh committed:
Merge branch 'master' into branch-A-CodeQuality

yanqiyqh committed:
Merge pull request #3 from yanqiyqh/branch-A-CodeQuality

Branch a code quality

yanqiyqh committed:
Add static variable DAYS_FOR_REMINDER

yanqiyqh committed:
Add method needsReminder() that returns a boolean indicating if the Deadline Task needs a reminder

yanqiyqh committed:
Add needsReminder() with javadoc

yanqiyqh committed:
Add javadoc for new method needsReminder()

yanqiyqh committed:
Add default needsReminder() method in Task and add @Override for Deadline and Event

yanqiyqh committed:
Add getTasksToRemind()

yanqiyqh committed:
Modify greet() to include reminders

yanqiyqh committed:
Fix bug that mapped from date as to date

yanqiyqh committed:
Modify greeting() to include more context on reminders

yanqiyqh committed:
Merge pull request #4 from yanqiyqh/branch-BCD-Extension

Branch bcd extension

yanqiyqh committed:
Fix Event bug

yanqiyqh committed:
Add bot name to GUI, add GUI sample

yanqiyqh committed:
Update README.md

yanqiyqh committed:
Update build.gradle file

yanqiyqh committed:
Update JUnit tests
